library/src/com/google/maps/android/clustering/ClusterManager.java

```diff
@@ -92,13 +93,23 @@ public void setAlgorithm(Algorithm<T> algorithm) {
             if (mAlgorithm != null) {
                 algorithm.addItems(mAlgorithm.getItems());
             }
-            mAlgorithm = new PreCachingAlgorithmDecorator<T>(algorithm);
+
+            mAlgorithm = algorithm;
         } finally {
             mAlgorithmLock.writeLock().unlock();
         }
+
+        if (mAlgorithm instanceof GoogleMap.OnCameraChangeListener) {
+            ((GoogleMap.OnCameraChangeListener) mAlgorithm).onCameraChange(mMap.getCameraPosition());
+        }
+
         cluster();
     }
 
+    public void setClusterOnlyVisibleArea(boolean onlyVisibleArea) {
```

Comment:
If we could use Java 8 we could add a default method to the interface, but alas that's not an option.

How about extending Algorithm (e.g. `ViewBasedAlgorithm`) and use that to determine instead. You may even be able to move the onCameraChangeListener code in there, too.